app/controllers/admin/events_controller.rb

```diff
@@ -2,26 +2,33 @@ class Admin::EventsController < Admin::ApplicationController
   include DateRange
 
   def index
-    if %w(views emails congress_messages tweets calls signatures).include? params[:type]
+    if params[:type].blank?
+      @data = events.group_by_type_in_range(start_date, end_date)
+      @columns = Ahoy::Event.action_types(action_page)
+    elsif Ahoy::Event.action_types.include? params[:type].to_sym
       @data = events.send(params[:type]).group_in_range(start_date, end_date)
     else
       render nothing: true, status: :bad_request
+      return
+    end
+    respond_to do |format|
+      format.html { render "table" }
+      format.json { render json: @data }
     end
-    render json: @data
-  end
-
-  def views
-    @actionPage = ActionPage.friendly.find(params[:action_page_id])
-    @views = @actionPage.events.views.group_in_range(start_date, end_date)
   end
 
-
   private
 
-  def events
+  def action_page
     if params[:action_page_id]
-      action_page = ActionPage.friendly.find(params[:action_page_id])
-      raise ActiveRecord::RecordNotFound unless action_page
+      @actionPage ||= ActionPage.friendly.find(params[:action_page_id])
+      raise ActiveRecord::RecordNotFound unless @actionPage
```

Comment:
We can remove this raise since it's redundant.

Reply:
oh awesome so much prettier, thanks!